pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/SSBQueryIntegrationTest.java

```diff
@@ -0,0 +1,265 @@
+import org.apache.pinot.client.ConnectionFactory;
+import org.apache.pinot.client.ResultSetGroup;
+import org.apache.pinot.common.datatable.DataTableFactory;
+import org.apache.pinot.core.common.datatable.DataTableBuilderFactory;
+import org.apache.pinot.query.service.QueryConfig;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.apache.pinot.spi.utils.CommonConstants;
+import org.apache.pinot.tools.BootstrapTableTool;
+import org.apache.pinot.util.TestUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+
+public class SSBQueryIntegrationTest extends BaseClusterIntegrationTest {
+  private static final Map<String, String> SSB_QUICKSTART_TABLE_RESOURCES = ImmutableMap.of(
+      "customer", "examples/batch/ssb/customer",
+      "dates", "examples/batch/ssb/dates",
+      "lineorder", "examples/batch/ssb/lineorder",
+      "part", "examples/batch/ssb/part",
+      "supplier", "examples/batch/ssb/supplier");
+
+  @BeforeClass
+  public void setUp()
+      throws Exception {
+    // Start the Pinot cluster
+    TestUtils.ensureDirectoriesExistAndEmpty(_tempDir, _segmentDir, _tarDir);
+
+    startZk();
+    startController();
+    startBroker();
+    startServer();
+
+    // Setting data table version to 4
+    DataTableBuilderFactory.setDataTableVersion(DataTableFactory.VERSION_4);
+
+    // load all SSB tables to both Pinot and H2.
+    loadSSBToPinot();
+
+    // load all SSB tables to H2 as well.
+    loadSSBToH2();
+  }
+
+  @Test(dataProvider = "ssbQueryDataProvider")
+  public void testSSBQueries(String query)
+      throws Exception {
+    testSSBQuery(query);
+  }
+
+  private void testSSBQuery(String query)
+      throws Exception {
+    // connection response
+    ResultSetGroup pinotResultSetGroup = getPinotConnection().execute(query);
+    org.apache.pinot.client.ResultSet resultTableResultSet = pinotResultSetGroup.getResultSet(0);
+    int numRows = resultTableResultSet.getRowCount();
+    int numColumns = resultTableResultSet.getColumnCount();
+
+    // h2 response
+    Assert.assertNotNull(_h2Connection);
+    Statement h2statement = _h2Connection.createStatement(ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+    h2statement.execute(query);
+    ResultSet h2ResultSet = h2statement.getResultSet();
+
+    // compare results.
+    if (h2ResultSet.first()) {
+      for (int i = 0; i < numRows; i++) {
+        for (int c = 0; c < numColumns; c++) {
+          String h2Value = h2ResultSet.getString(c + 1);
+          String pinotValue = resultTableResultSet.getString(i, c);
+          boolean error = ClusterIntegrationTestUtils.fuzzyCompare(h2Value, pinotValue, pinotValue);
+          if (error) {
+            throw new RuntimeException("Value: " + c + " does not match, expected: " + h2Value
+                + ", got pinot value: " + pinotValue);
```

Comment:
can we also indicate which query and which row/column it was? perhaps even print out the entire row.